app/services/tool_filter_service.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+class ToolFilterService
+  def initialize(params)
+    @params = params
+  end
+
+  def call
+    tool_groups = []
+    ToolGroup.all.each { |tool_group| tool_groups << tool_group if match_params(tool_group, @params) }
+    grouped_array = group_resources(tool_groups)
+    order_results(grouped_array)
+  end
+
+  private
+
+  def order_results(grouped_array)
+    result = grouped_array.map do |key, values|
+      counter = values.size
+      sum = values.sum { |o| o[:tool_group_suggestions_weight] * o[:resource_tool_group_suggestions_weight] }
+      average = sum / counter
+      [Resource.find(key), average]
+    end
+
+    sorted_result = result.sort_by { |subarray| -subarray[1] }
+    sorted_result.map { |subarray| subarray[0] }
+  end
+
+  def group_resources(tool_groups)
+    result = tool_groups.flat_map do |tool_group|
+      tool_group.resource_tool_groups.map do |resource|
+        {
+          tool_group_suggestions_weight: tool_group.suggestions_weight,
+          resource_id: resource.resource_id,
+          resource_tool_group_suggestions_weight: resource.suggestions_weight
+        }
+      end
+    end
+
+    result.group_by { |hash| hash[:resource_id] }
+  end
+
+  def match_params(tool_group, params)
+    country = params["filter"]["country"]&.upcase
```

Comment:
NoMethodError: undefined method `[]' for nil:NilClass

https://app.rollbar.com/a/Cru/fix/item/mobile-content-api/319

```
curl --location 'https://mobile-content-api-stage.cru.org/resources/suggestions'
```

Reply:
<img width="763" alt="Screenshot 2023-08-01 at 4 56 40 PM" src="https://github.com/CruGlobal/mobile-content-api/assets/47818/520462a7-1baf-4804-a9ce-cbf35eac5390">